include/ocpp/v201/component_state_manager.hpp

```diff
@@ -23,12 +23,102 @@ struct FullConnectorStatus {
 
     /// \brief Translates the individual state to an Available/Unavailable/Occupied/Reserved/Faulted state
     /// This does NOT take into account the state of the EVSE or CS,
-    /// and is intended to be used internally by the ComponentStateManager.
+    /// and is intended to be used internally by the ComponentStateManagerInterface.
     ConnectorStatusEnum to_connector_status();
 };
 
+class ComponentStateManagerInterface {
+public:
+    virtual ~ComponentStateManagerInterface();
+
+    virtual void set_cs_effective_availability_changed_callback(
+        const std::function<void(const OperationalStatusEnum new_status)>& callback) = 0;
+
+    /// \brief Set a callback to be called when the effective Operative/Inoperative state of an EVSE changes.
```

Comment:
I see we have duplicates now of the documentation. Is it an idea to only add documentation to either the interface or the implementation to keep it DRY?

Reply:
Removed the comments from the implementation.